internal/formats/common/cyclonedxhelpers/decoder.go

```diff
@@ -45,8 +45,8 @@ func GetDecoder(format cyclonedx.BOMFileFormat) sbom.Decoder {
 }
 
 func toSyftModel(bom *cyclonedx.BOM) (*sbom.SBOM, error) {
-	if bom == nil {
```

Comment:
Thanks for adding the nil check! 

Does it make sense here to error on no components?

https://github.com/cpendery/syft/blob/7d24424b58bc78e2fa354c76466a90dbd3ac88ac/internal/formats/common/cyclonedxhelpers/decoder.go#L34-L39

If we constructed a bom initially with an empty list of components we would not see the nil pointer here and coule return/cover cases to users where no comonents were found.

If we instead

Reply:
That makes sense! I updated the code to reflect the suggestion